packages/maskbook/src/web3/UI/EthereumIfWalletConnected.tsx

```diff
@@ -0,0 +1,55 @@
+import { createStyles, Grid, makeStyles } from '@material-ui/core'
+import { useCallback } from 'react'
+import ActionButton from '../../extension/options-page/DashboardComponents/ActionButton'
+import { WalletMessages } from '../../plugins/Wallet/messages'
+import { useRemoteControlledDialog } from '../../utils/hooks/useRemoteControlledDialog'
+import { useI18N } from '../../utils/i18n-next-ui'
+import { useAccount } from '../hooks/useAccount'
+import { useChainIdValid } from '../hooks/useChainState'
+
+const useStyles = makeStyles((theme) =>
+    createStyles({
+        button: {},
+    }),
+)
+
+export interface EthereumIfWalletConnectedProps {
+    children?: React.ReactNode
+}
+
+export function EthereumIfWalletConnected(props: EthereumIfWalletConnectedProps) {
```

Comment:
I have to say this is super strange

Reply:
I added a usage section. Any better name for this?